### PR TITLE
load cloned world with doLoad() instead of addWorld() (fixes #2378)

### LINKED ISSUE #2378: Cloning world initially applies default properties to cloned world
### Steps to reproduce
- Create a world named "original"
- Disallow monsters with `mvm set monsters false original`
- Clone the world with `mv clone original clone`
- Teleport to the clone with `mvtp clone`
- Spawn a zombie with a spawn egg

### Expected behaviour
The zombie does not appear in the world "clone", as the world "original" had monsters set to false.

### Actual behaviour
The zombie does appear in the world "clone".

### Further notes
- Having a look into `worlds.yml` directly after cloning verifies that the default properties are set
- `mvinfo clone` prints the correct properties, but they are actually not applied
- Restarting the server fixes the problem

### Cause
In `WorldManager.java` around line 200 the cloned world is added with `addWorld()`. This creates a set of default properties. After the `addWorld()` call, the original properties are copied to the properties map of the cloned world, but the cloned world does not apply them.

### Fix
I created a patch fixing this problem. Expect the pull request in a couple of minutes.

### PR DESCRIPTION
**Pull request by rti**
This pull request fixes #2378 

We actually can skip all the `WorldCreator` setup done by `addWorld()` as the cloned world is ready and just needs to be loaded. Just using `doLoad()` enables us to cleanly apply the copied properties, which is the intention of this patch.

### CODE REVIEW

File: src/main/java/com/onarandombox/MultiverseCore/utils/WorldManager.java [hunk 1]
```diff
@@ -193,16 +193,31 @@ public boolean cloneWorld(String oldName, String newName) {
...
+            // initialize new properties with old ones
+            WorldProperties newProps = new WorldProperties();
+            newProps.copyValues(worldsFromTheConfig.get(oldName));
+            // don't keep the alias the same -- that would be useless
+            newProps.setAlias("");
+            // store the new properties in worlds config map
+            worldsFromTheConfig.put(newName, newProps);
+
+            // save the worlds config to disk (worlds.yml)
+            if(!saveWorldsConfig()) {
```
Comment (nicegamer7) [inline src/main/java/com/onarandombox/MultiverseCore/utils/WorldManager.java:206]: Please add a space after `if`.

File: src/main/java/com/onarandombox/MultiverseCore/utils/WorldManager.java [hunk 2]
```diff
@@ -193,16 +193,31 @@ public boolean cloneWorld(String oldName, String newName) {
...
+                this.plugin.log(Level.SEVERE, "Failed to save worlds.yml");
+                return false;
+            }
+
+            // actually load the world
+            if(doLoad(newName)) {
```
Comment (nicegamer7) [inline src/main/java/com/onarandombox/MultiverseCore/utils/WorldManager.java:212]: Please add a space after `if`.

File: src/main/java/com/onarandombox/MultiverseCore/utils/WorldManager.java [hunk 3]
```diff
@@ -193,16 +193,31 @@ public boolean cloneWorld(String oldName, String newName) {
...
+               Logging.fine("Succeeded at loading cloned world '" + newName + "'");
```
Comment (nicegamer7) [inline src/main/java/com/onarandombox/MultiverseCore/utils/WorldManager.java:213]: Just for consistency with the the modern MV codebase, this should be changed to from
```java
Logging.fine(...
```
to
```java
this.plugin.log(Level.FINE, ...
```

**Review by nicegamer7 — CHANGES_REQUESTED** [review]
Thanks for the PR! Overall seems good. I just requested some style changes, nothing big.

File: src/main/java/com/onarandombox/MultiverseCore/utils/WorldManager.java [hunk 4]
```diff
@@ -193,16 +193,31 @@ public boolean cloneWorld(String oldName, String newName) {
...
-                newWorld.setAlias("");
-                return true;
+
+            // initialize new properties with old ones
+            WorldProperties newProps = new WorldProperties();
+            newProps.copyValues(worldsFromTheConfig.get(oldName));
+            // don't keep the alias the same -- that would be useless
+            newProps.setAlias("");
+            // store the new properties in worlds config map
+            worldsFromTheConfig.put(newName, newProps);
```
Comment (nicegamer7) [inline src/main/java/com/onarandombox/MultiverseCore/utils/WorldManager.java:203]: This is also a really small nitpick, but when referring to `worldsFromTheConfig`, please use `this.worldsFromTheConfig` as it makes it easier to see that it's an attribute of the class.

**Comment by rti** [comment]
Thanks a lot for the review. I addressed your comments.

**Comment by rti** [comment]
Not sure whether force-pushing was a good idea here. Github seems to be a bit confused now. Sorry. Hope that works for you.

File: src/main/java/com/onarandombox/MultiverseCore/utils/WorldManager.java [hunk 5]
```diff
@@ -193,16 +193,31 @@ public boolean cloneWorld(String oldName, String newName) {
...
-                Logging.fine("Succeeded at importing world");
-                MVWorld newWorld = (MVWorld) this.getMVWorld(newName);
-                newWorld.copyValues(this.worldsFromTheConfig.get(oldName));
-                // don't keep the alias the same -- that would be useless
-                newWorld.setAlias("");
-                return true;
+
+            // initialize new properties with old ones
+            WorldProperties newProps = new WorldProperties();
+            newProps.copyValues(worldsFromTheConfig.get(oldName));
```
Comment (nicegamer7) [inline src/main/java/com/onarandombox/MultiverseCore/utils/WorldManager.java:199]: You missed a `this.` over here.

**Review by nicegamer7 — CHANGES_REQUESTED** [review]

**Comment by nicegamer7** [comment]
Force pushing is fine.

File: src/main/java/com/onarandombox/MultiverseCore/utils/WorldManager.java (hunk 5 above)
Comment (rti) [reply src/main/java/com/onarandombox/MultiverseCore/utils/WorldManager.java:199]: Oops. Thanks for pointing that out.

**Comment by benwoo1110** [comment]
Also just a note, usually it's good to create a new branch when making a PR instead of using master.

**Comment by rti** [comment]
You are right. Would have been way cleaner. Though, I created my fork only to create this PR.

**Comment by rti** [comment]
Hm, the travis build stopped while downloading something. Is there a way to re-trigger the build?

**Review by benwoo1110 — APPROVED** [review]

**Review by nicegamer7 — APPROVED** [review]

**Comment by nicegamer7** [comment]
I figured out how to trigger a restart 👍.

**Comment by dumptruckman** [comment]
Awesome. Thank you for this!

**Pull request merged**